joblib/func_inspect.py

```diff
@@ -83,7 +83,7 @@ def _clean_win_chars(string):
         # In Python 3, quote is elsewhere
         import urllib.parse
         quote = urllib.parse.quote
-    for char in ('<', '>', '!', ':', '\\'):
+    for char in ('<', '>', '!', ':'):
```

Comment:
this change broke one of the tests but fix the filename too long issue on appveyor.